packages/marketplace/src/components/ui/app-detail/app-content/app-content.tsx

```diff
@@ -0,0 +1,169 @@
+          <div className="control has-icons-right">
+            <input className="input is-primary" id="apiKey" type="text" name="apiKey" value={apiKey || ''} readOnly />
+            <span className="icon is-right">
+              <FaCopy />
+            </span>
+          </div>
+        </CopyToClipboard>
+      )}
+    </>
+  )
+}
+
+const AppContent: React.FC<AppContentProps> = ({ appDetailData, loginType }) => {
+  const {
+    externalId,
+    isListed,
+    id,
+    developer,
+    authFlow,
+    isWebComponent,
+    apiKey,
+    media = [],
+    scopes = [],
+    description = '',
+  } = appDetailData
+  const [isShowApiKey, setIsShowApikey] = React.useState<boolean>(false)
+
+  const isCurrentLoggedUserDeveloper = loginType === 'DEVELOPER'
+  const carouselImages = media.filter(({ type }) => type === 'image')
+  const settings: Settings = {
+    dots: false,
+    speed: 500,
+    variableWidth: true,
+    prevArrow: (
+      // @ts-ignore
+      <SlickButtonNav>
+        <ChevronLeftIcon />
+      </SlickButtonNav>
+    ),
+    nextArrow: (
+      // @ts-ignore
+      <SlickButtonNav>
+        <ChevronLeftIcon />
+      </SlickButtonNav>
+    ),
+  }
+
+  return (
+    <Grid>
+      <GridItem className="is-4">
+        {isCurrentLoggedUserDeveloper && (
+          <>
+            <div key="app-developer" className={styles.appInfoRow}>
+              <p className={styles.appInfoProperty}>Developer:</p>
+              <p>{developer}</p>
+            </div>
+            <div key="app-id" className={styles.appInfoRow}>
+              <p className={styles.appInfoProperty}>Client ID:</p>
+              <p>{externalId}</p>
+            </div>
+            <div key="app-listed" className={styles.appInfoRow}>
+              <p className={styles.appInfoProperty}>Status:</p>
+              <p className={styles.appInfoSpace}>{isListed ? 'Listed' : 'Not listed'}</p>
+              {isListed ? <FaCheck className={styles.isListed} /> : <FaTimes className={styles.notListed} />}
+            </div>
+            {authFlow === AuthFlow.CLIENT_SECRET && id && <AppAuthenticationDetail appId={id} />}
+          </>
+        )}
+        {renderShowApiKeyForWebComponent({
+          isWebComponent,
+          isShowApiKey,
+          setIsShowApikey,
+          apiKey: apiKey,
+          isCurrentLoggedUserDeveloper,
+        })}
+      </GridItem>
+      <GridItem className="is-8">
+        <HTMLRender className={styles.description} html={description} />
+        <br />
+        {carouselImages.length > 0 && (
+          <div className={carouselStyles.container}>
+            <Slider {...settings}>
+              {carouselImages.map(({ uri }, index) => (
+                <div key={index} className={carouselStyles.slide}>
+                  <img src={uri} />
+                </div>
+              ))}
+            </Slider>
+          </div>
+        )}
+        <br />
+        <Grid isMultiLine>
+          {scopes.map(item => (
```

Comment:
I think this component too long pls separate to function `renderScopes`

Reply:
Hi @tanphamhaiduong,
Because this view is going to be changed when building the new UI for standalone app page, refactoring these codes here will means nothing, we will still have to rework anyway